error: 'QUICNetVConnection::free' hides overloaded virtual function [-Werror,-Woverloaded-virtual]

```
  CXX      QUICStatelessRetry.o
In file included from QUICStatelessRetry.cc:30:
/Users/masaori/src/gitlab.sd.apple.com/trafficserver/trafficserver-quic/iocore/net/P_QUICNetVConnection.h:172:8: error: 'QUICNetVConnection::free' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
  void free(EThread *t) override;
       ^
../../../lib/ts/Ptr.h:84:3: note: hidden overloaded virtual function 'RefCountObj::free' declared here: different number of parameters (0 vs 1)
  free()
  ^
1 error generated.
```

#3140 has a warning by clang.

--- a/iocore/net/P_QUICNetVConnection.h
+++ b/iocore/net/P_QUICNetVConnection.h
@@ -170,6 +170,7 @@ public:
   void start();
   void remove_connection_ids();
   void free(EThread *t) override;
+  void free() override;
   void destroy(EThread *t);
 
   UDPConnection *get_udp_con();

--- a/iocore/net/QUICNetVConnection.cc
+++ b/iocore/net/QUICNetVConnection.cc
@@ -235,6 +235,12 @@ QUICNetVConnection::free(EThread *t)
   this->_packet_handler->close_conenction(this);
 }
 
+void
+QUICNetVConnection::free()
+{
+  this->free(this_ethread());
+}
+
 // called by ET_UDP
 void
 QUICNetVConnection::remove_connection_ids()

